service/src/MSRIOGroup.cpp

```diff
@@ -1260,7 +1261,20 @@ namespace geopm
             for (const auto &file : files) {
                 std::string filename = dir + "/" + file;
                 if (string_begins_with(file, "msr_") && string_ends_with(file, ".json")) {
-                    data_files.insert(filename);
+                    bool do_insert = true;
+                    int fd = open(filename.c_str(), O_RDONLY);
+                    if (fd < 0) {
+                        std::cerr << "Warning: <geopm> Failed to open() MSR plugin: "
+                                  << strerror(errno) << std::endl;
+                        do_insert = false;
+                    }
+                    else {
+                        // TODO: Insert fd into tracking vector to close on destruction
```

Comment:
This TODO is actually more difficult than I want it to be.  This function is static and is used in geopmadmin to generate the msr-safe allowlist.  Not sure if it's even worth it as the fd's will be close when the process ends.